### PR TITLE
feat(ui): added testing modelu for components

### PR DESCRIPTION
**Pull request by fmayorgad**

### CODE REVIEW

File: src/Components/Counter/Counter.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { screen, render, fireEvent } from '@testing-library/react';
+import { Counter } from './Counter';
+
+describe('Test Counter Component', () => {
```
Comment (jgmurillo10) [inline src/Components/Counter/Counter.test.js:5]: nit: you could validate as well for the increase and decrease buttons being render. Check for the existence and label :)

File: src/Components/GenreSelector/GenreSelector.js [hunk 2]
```diff
@@ -19,11 +19,11 @@ export class GenreSelector extends React.Component {
     render() {
         return (
             <div className='genreSelectContainer'>
-                <li>
+                <ol role="tablist">
```
Comment (jgmurillo10) [inline src/Components/GenreSelector/GenreSelector.js]: Nice improvement 🔥

File: src/Components/GenreSelector/GenreSelector.test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { screen, render, fireEvent } from '@testing-library/react';
+import { GenreSelector } from './GenreSelector';
+
+describe('Test GenreSelector Component', () => {
+
+    const genreList = [
+        { id: 0, name: 'all' },
```
Comment (jgmurillo10) [inline src/Components/GenreSelector/GenreSelector.test.js:8]: tip: you don't necessary need to define the same list in the tests. You could have something like:
```
const genreList = [ {  id: 0, name: 'test name' } ];
```

File: src/Components/GenreSelector/GenreSelector.test.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        { id: 2, name: 'terror' },
+        { id: 3, name: 'comedy' },
+        { id: 4, name: 'documentary' },
+        { id: 5, name: 'crime' },
+    ];
+
+    const selectedGenre = 0;
+
+    const onSelectGenre = (e) => {
+        console.log(e);
```
Comment (jgmurillo10) [inline src/Components/GenreSelector/GenreSelector.test.js]: You don't need console logging on tests. They are helpful in while developing but production code does not need logs. You can simply do:
```
const onSelectGenre = jest.fn();
...
render(<GenreSelector  onSelectGenre={onSelectGenre} genreList={genreList} />)
```

File: src/Components/GenreSelector/GenreSelector.js (hunk 2 above)
Comment (jgmurillo10) [inline src/Components/GenreSelector/GenreSelector.js]: You can use the role for selection on test but another "good practice" might be using a `data-test-id` attribute with a descriptive value
```
<o data-test-id="genre-list-container">
```

File: src/Components/GenreSelector/GenreSelector.test.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    test('renders all genres passedin props', () => {
+        render(<GenreSelector selectedGenre={selectedGenre} genreList={genreList} onSelectGenre={onSelectGenre} />);
+        const counterLabel = screen.getAllByRole('tab');
+        expect(counterLabel.length).toEqual(genreList.length);
+    })
+
+    test('highlights a selected genre passed in props', () => {
+        render(<GenreSelector selectedGenre={selectedGenre} genreList={genreList} onSelectGenre={onSelectGenre} />);
+        const counterLabel = screen.getAllByRole('tab');
+        const selectedItem = counterLabel.filter(c => c.classList.contains('active')).reduce(c => c);
```
Comment (jgmurillo10) [inline src/Components/GenreSelector/GenreSelector.test.js:30]: This fills a little bit "hacky". The `render` function returns an object with a `container` attribute that you can query:
```
const { container } = render(<GenreSelector selectedGenre={selectedGenre} genreList={genreList} onSelectGenre={onSelectGenre} />);
...
const selectedItem = container.getElementsByClassName('active')[0];
```

Comment (jgmurillo10) [reply src/Components/GenreSelector/GenreSelector.test.js:30]: Also, because you already know which button should have the class, you could get the button using screen and check for classname
```
// pseudo code
const selectedItem = screen.getByText(`button label`);
expect(counterLabel.classList.contains('active')).toBe(true);
```

File: src/Components/GenreSelector/GenreSelector.test.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        expect(parseInt(selectedItem.id)).toEqual(selectedGenre);
+    })
+
+    test('click event on a genre button component calls "onChange" callback and passes correct genre in arguments', () => {
```
Comment (jgmurillo10) [inline src/Components/GenreSelector/GenreSelector.test.js:34]: It's recommended to check the params for the select genre rather than console logging :)

File: src/Components/SearchForm/SearchForm.test.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { screen, render, fireEvent } from '@testing-library/react';
+import { SearchForm } from './SearchForm';
+
+describe('Test SearchForm Component', () => {
+
+    const searchPassedtoComponent = 'Drama';
+
+    const onSearch = (input) => {
```
Comment (jgmurillo10) [inline src/Components/SearchForm/SearchForm.test.js]: Same here, let's use `jest.fn`

File: src/Components/SearchForm/SearchForm.test.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        });
+
+        const handleKeyDown = jest.fn((e) => {
+            console.log(e.target.value)
+            if (e.keyCode === 13) {
+                handleInputChange(e.target.value);
+            }
+        });
+
+        render(<div className='searchFormContainer'>
```
Comment (jgmurillo10) [inline src/Components/SearchForm/SearchForm.test.js]: Instead of rendering this components which are copy-paste of the `SearchForm` it might be a better approach to spy on the onSearch param function
```
const onSearchSpy = jest.fn();
render(<SearchForm onSearch={onSearchSpy} searchQuery={searchQuery} />);
//then check for onSearch called with
expect(onSearchSpy).toHaveBeenCalledWith(newValue);
```

File: src/Components/GenreSelector/GenreSelector.test.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        { id: 5, name: 'crime' },
+    ];
+
+    const selectedGenre = 0;
+
+    const onSelectGenre = (e) => {
+        console.log(e);
+    }
+
+    test('renders all genres passedin props', () => {
```
Comment (jgmurillo10) [inline src/Components/GenreSelector/GenreSelector.test.js:21]: Might be interesting to add a test where the list is empty 🤔 
```
test('render message when list is empty');
```
or something like that

File: babel.config.js
Comment (jgmurillo10) [inline babel.config.js:1]: All of these cypress/**/* files are intended to be committed or should be git ignored?

File: src/App.js
Comment (jgmurillo10) [inline src/App.js:1]: The App test `src/App.test.js` looks a little bit outdated. Would you mind removing it,  or changing it to match the purpose of the file? e.g. `screen` is being imported but it's unused.

**Review by jgmurillo10 — CHANGES_REQUESTED** [review]
Amazing work on these tests, I left you a couple of comments

File: src/Components/SearchForm/SearchForm.test.js [hunk 10]
```diff
@@ -6,9 +6,8 @@ describe('Test SearchForm Component', () => {
 
     const searchPassedtoComponent = 'Drama';
 
-    const onSearch = (input) => {
-        console.log(input);
-    }
+    const onSearch = jest.fn((e) => {
```
Comment (jgmurillo10) [inline src/Components/SearchForm/SearchForm.test.js:9]: nit: 
```
 const onSearch = jest.fn();
```

File: src/Components/GenreSelector/GenreSelector.test.js [hunk 11]
```diff
@@ -16,32 +16,31 @@ describe('Test GenreSelector Component', () => {
     const selectedGenre = 0;
 
     const onSelectGenre = (e) => {
```
Comment (jgmurillo10) [inline src/Components/GenreSelector/GenreSelector.test.js:18]: nit:

const onSelectGenre = jest.fn();

**Review by jgmurillo10 — APPROVED** [review]
Great work! LGTM
note: there are some spaces added that might be removable with the help of linters (for production work)
